grin/src/miner.rs

```diff
@@ -527,22 +510,45 @@ impl Miner {
 			} else {
 				debug!(
 					LOGGER,
-					"setting pubkey in miner to pubkey from block_fees - {:?}",
-					block_fees
+					"setting pubkey in miner to pubkey from block_fees - {:?}", block_fees
 				);
 				key_id = block_fees.key_id();
 			}
 		}
 	}
 
+        // Ensure a block is built and returned
```

Comment:
I think this should be externalized from the miner into its own thing as it's a generally useful facility. Just add the chain and pool (needed by `build_block`) as an argument and put it into a `mine_block.rs` or something similar.

Reply:
I think I understood this request, and made an attempt at it.
Again, please see the changes in the new testnet2 pull request:
https://github.com/mimblewimble/grin/pull/937